src/HttpCache/EventListener/AddTagsListener.php

```diff
@@ -44,18 +44,22 @@ final class AddTagsListener
     private $xkeyEnabled;
     private $xkeyGlue;
     private $httpTagsEnabled;
+    private $header;
+    private $separator;
 
-    public function __construct(IriConverterInterface $iriConverter, ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory = null, bool $xkeyEnabled = false, string $xkeyGlue = ' ', bool $httpTagsEnabled = true)
+    public function __construct(IriConverterInterface $iriConverter, ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory = null, bool $xkeyEnabled = false, string $xkeyGlue = ' ', bool $httpTagsEnabled = true, string $header = 'Cache-Tags', string $separator = ',')
```

Comment:
Couldn't we do something like this instead (the real implementation will of course be more complicated because we need a BC layer)?

```suggestion
public const VARNISH_HEADER_NAME = 'xkey';
public const VARNISH_SEPARATOR = ' ';
public const SOUIN_HEADER_NAME = 'Cache-Tag';
public const SOUIN_SEPARATOR = ',';

    public function __construct(IriConverterInterface $iriConverter, ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory = null, string $headerName = false, string $separator = ' ') {}
    
    public static function createSouinInstance(IriConverterInterface $iriConverter, ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory): static {};
    public static function createVarnishInstance(IriConverterInterface $iriConverter, ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory): static {};
```

Reply:
I don't think this is a good approach and is more complex than it should be. All additional parameters are not mandatory, they don't break anything. It uses the existing parameters values for the cache header and the separator.